analytics/views.py

```diff
@@ -1011,6 +1021,70 @@ def get_activity(request: HttpRequest) -> HttpResponse:
         context=dict(data=data, title=title, is_home=True),
     )
 
+@require_server_admin
+def support(request: HttpRequest) -> HttpResponse:
+    context = {}  # type: Dict[str, Any]
+    if request.method == "POST":
+        realm_id = request.POST.get("realm_id", None)
+        realm = Realm.objects.get(id=realm_id)
+
+        new_plan_type = request.POST.get("plan_type", None)
+        if new_plan_type is not None:
+            new_plan_type = int(new_plan_type)
+            current_plan_type = realm.plan_type
+            if new_plan_type != current_plan_type:
```

Comment:
this should be do_change_plan_type, and should be called regardless of whether new_plan_type == current_plan_type.